Assignment6.Rmd

```diff
@@ -48,22 +48,23 @@ post(c.tree1, file = "tree1.ps", title = "MOOC") #This creates a pdf image of th
 #If we are worried about overfitting we can remove nodes form our tree using the prune() command, setting cp to the CP value from the table that corresponds to the number of nodes we want the tree to terminate at. Let's set it to two nodes.
 
 ```{r}
-c.tree2 <- prune(c.tree1, cp = )#Set cp to the level at which you want the tree to end
+c.tree2 <- prune(c.tree1, cp=printcp(c.tree1)[2,1])
+#Set cp to the level at which you want the tree to end
 
 #Visualize this tree and compare it to the one you generated earlier
 
-post(c.tree2, file = "tree2.ps", title = "MOOC") #This creates a pdf image of the tree
+rpart.plot(c.tree2,type=2,extra = 104,fallen.leaves = TRUE, main="Decision Tree")
+#This creates a pdf image of the tree
 ```
 
 #Now use both the original tree and the pruned tree to make predictions about the the students in the second data set. Which tree has a lower error rate?
 
 ```{r}
 M2$predict1 <- predict(c.tree1, M2, type = "class")
-
 M2$predict2 <- predict(c.tree2, M2, type = "class")
-
-table(M2$certified, M2$predict1)
-
-table(M2$certified, M2$predict2)
-
+#prediction accuracy of original tree:
+sum(diag(table(M2$certified, M2$predict1)))/nrow(M2)
+#prediction accuracy of pruned tree
+sum(diag(table(M2$certified, M2$predict2)))/nrow(M2)
+#form the results we could see that pruned tree has much higher precision
```

Comment:
Can you provide a summary of your findings? Which model was more accurate and why do you think that is?